### DISCUSSION TITLE
different street name in OsmAnd and OSM

### DISCUSSION THREAD

**Discussion by benasocj** [osmandapp/OsmAnd#12602] [Q&A] [answered]
OSM says that the street name is "Grazerstraße 12": https://www.openstreetmap.org/way/351191479

OsmAnd however, says that it is "Grazer Straße 12":
![osmand](https://user-images.githubusercontent.com/986086/130954396-82ef7488-b338-4291-92a0-b3ef037e4c90.jpg)

My downloaded OsmAnd offline map is up-to-date and I didn't notice any recent changes on OSM that wouldn't be included in the offline map yet.

Can someone please explain why the address is different?

**Comment by alisa911** [comment]
Hi! The reason for this bug is in the algorithms of building maps in OsmAnd. For example, in this case, next to this address there is a big relation with a similar name (Grazer Straße) https://www.openstreetmap.org/relation/962463#map=10/46.9380/15.4827&layers=C. 
Therefore, the algorithm decides to rename street, since it has a higher internal rating (has more nodes).
In any case, this is one and the same street, which was named differently within different administrative offices. Perhaps this algorithm will be revised in the future.

**Comment by vshcherb** [comment, marked as answer]
We've identified more precise answer to it. In this city there is a street with similar name Grazer Straße (https://www.openstreetmap.org/way/26565000) cause it's connected to 'GrazerStraße' and has similar name they've been merged to 1 street in Address search and produce same result. 

On the map there are 2 different names cause they're stored and processed separately as each segment. In Address search it's important to merge so user could see proper list of streets in the city.
